src/backend/direct.rs

```diff
@@ -84,15 +84,13 @@ mod pass_impl {
                 )
             }
         }
-        fn set_push_constants(&mut self, offset: u32, data: &[u32]) {
+        fn set_push_constants(&mut self, offset: u32, data: &[u8]) {
             unsafe {
                 wgpu_compute_pass_set_push_constant(
                     self,
                     offset,
-                    (data.len() * std::mem::size_of::<u32>())
-                        .try_into()
-                        .unwrap(),
-                    data.as_ptr(),
+                    data.len().try_into().unwrap(),
+                    data.as_ptr() as *const u32,
```

Comment:
This api should be changed in wgpu-core to take a pointer to u8. As of right now this is unsound as the data may not be aligned to 4 bytes, causing ub on dereference.